Makefile

```diff
@@ -0,0 +1,264 @@
+DIR := ${CURDIR}
+
+E:=@
+ifeq ($(V),1)
+	E=
+endif
+
+cyan := $(shell which tput > /dev/null && tput setaf 6 2>/dev/null || echo "")
+reset := $(shell which tput > /dev/null && tput sgr0 2>/dev/null || echo "")
+bold  := $(shell which tput > /dev/null && tput bold 2>/dev/null || echo "")
+
+.PHONY: default
+
+default: generate
+
+help:
+	@echo "$(bold)Usage:$(reset) make $(cyan)<target>$(reset)"
+	@echo "  $(cyan)generate$(reset)                              - generate protocol buffers and plugin interface code"
+	@echo "  $(cyan)generate-check$(reset)                        - ensure generated code is up to date"
+	@echo "  $(cyan)protogen$(reset)                              - compile protocol buffers"
+	@echo "  $(cyan)protogen-check$(reset)                        - ensure generated protocol buffers are up to date"
+	@echo "  $(cyan)plugingen$(reset)                             - generate plugin interface code"
+	@echo "  $(cyan)plugingen-check$(reset)                       - ensure generated plugin interface code is up to date"
+	@echo
+	@echo "For verbose output set V=1"
+	@echo "  for example: $(cyan)make V=1 build$(reset)"
```

Comment:
Since `build` is not a target in this Makefile, we may mention a valid target here.

Reply:
:+1, also cleaned up some of those targets that don't exist